### PR TITLE
ParserUtil: parseX(...) methods should not take in Optional #523

### LINKED ISSUE #523: Update ParserUtil#parseX methods
`ParserUtil#parseX(Optional<String>)` methods currently take in an `Optional<String>`. This violates the usage of Java Optional.

These methods are only used in AddCommandParser and EditCommandParser: 
**AddCommandParser**
```
Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).get();
...
```
We can change it to:
<pre>
<strike>Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).orElse(null));</strike>
Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).orElse(null)).get();
...
</pre>
**EditCommandParser**
```
ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).ifPresent(editPersonDescriptor::setName);
...
```
One of the possible ways to change it is:
<pre>
<strike>Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).orElse(null));</strike>
ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).orElse(null)).ifPresent(editPersonDescriptor::setName);
...
</pre>
~~Then, we have a new constructor for `EditPersonDescriptor` that takes in `Name`, `Phone` etc.~~

**ParserUtil#parseName(String)** will now look something like
<pre>
<strike>return name != null ? new Name(name) : null;</strike>
return name != null ? new Optional.of(Name(name)) : Optional.empty();
</pre>

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Fixes #523.

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/1] [ParserUtil: parseX\(\.\.\.\) methods should not take in Optional](https://github.com/se-edu/addressbook-level4/pull/556/commits/3b75fefd052dfcadfad46a290ff21e5d146d0852)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/6422e8f3048fc63e9e46a8d7937eb3bc433e74ad...3b75fefd052dfcadfad46a290ff21e5d146d0852))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/556/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 1]
```diff
@@ -64,101 +64,77 @@ public void parseIndex_validInput_success() throws Exception {
...
-
-    @Test
     public void parseEmail_invalidValue_throwsIllegalValueException() throws Exception {
         thrown.expect(IllegalValueException.class);
-        ParserUtil.parseEmail(Optional.of(INVALID_EMAIL));
+        ParserUtil.parseEmail(INVALID_EMAIL);
     }
 
     @Test
     public void parseEmail_optionalEmpty_returnsOptionalEmpty() throws Exception {
```
Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java]: Should be `..._null_...`

**Comment by CanIHasReview** [comment]
# v2

@Zhiyuan-Amos submitted v2 for review.

* [v2 1/1] [ParserUtil: parseX\(\.\.\.\) methods should not take in Optional](https://github.com/se-edu/addressbook-level4/pull/556/commits/6b204cbfcc41588673a7896599f7f56bb636a90e)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/6422e8f3048fc63e9e46a8d7937eb3bc433e74ad...6b204cbfcc41588673a7896599f7f56bb636a90e)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/2d100020383df5311354ba2d073ab5e5723a5405~1...2d100020383df5311354ba2d073ab5e5723a5405))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/556/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
@damithc would like to check if the architecture is fine :)

**Comment by damithc** [comment]
Remind me to discuss tomorrow.

**Comment by yamgent** [comment]
As discussed, ParserUtil methods are like the middle-man, so it is okay to use `Optional<?>` as a parameter here even though it is not the correct Java convention (convention = only use in return type).

**Comment by Zhiyuan-Amos** [comment]
Superseded by #523

**Pull request closed without merging**